enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,964 @@
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is by managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) along with a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and as a by product has significantly burdened the `opm` tool with feature requests to add an increasing number of sub-commands to allow index authors to manipulate their indexes in various ways. 
+
+## Goals
+
+A human consumable representation of packages in an index in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bundle/s from a channel, switch bundle/s to a different channel, edit the upgrade graph in a channel etc).
+2. Allow modification of upgrade graph for a package without needing to rebuild and republish the index the package belongs to(i.e combine packages from two indexes, create multiple indexes with packages from a single index, remove bundles from the index, switch channels bundles belong to etc).
+3. Allow for validation of the structure of an index against a standard definition of an index.
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+## Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+## User Stories 
+
+### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
```

Comment:
I was shooting for the tooling aspects to be a part of the [opm content management enhancement](https://github.com/operator-framework/enhancements/pull/26)

Reply:
Also

>One piece that seems missing is a tool to go from bundle image -> bundle json. index add would have to do that underneath, but it seems useful to get the the translation without automatically adding it to the index.

Does this sound like something that can be addressed in that enhancement instead of this? Or do you see a more immediate need for this?